ros2-library/src/main/java/us/ihmc/ros2/ROS2NodeBuilder.java

```diff
@@ -0,0 +1,438 @@
+         Properties properties = new Properties();
+         try (FileInputStream inputStream = new FileInputStream(networkParametersFile))
+         {
+            properties.load(inputStream);
+
+            if (properties.getProperty(networkParametersKey) != null)
+            {
+               possibleValues.push(Map.entry("(IHMCNetworkParameters.ini) " + networkParametersKey, properties.getProperty(networkParametersKey)));
+            }
+         }
+         catch (IOException e)
+         {
+            if (!(e instanceof FileNotFoundException))
+               LogTools.error(e);
+         }
+      }
+
+      if (!possibleValues.empty())
+      {
+         StringJoiner printout = new StringJoiner(" <- ");
+         for (int i = possibleValues.size() - 1; i >= 0; i--)
+         {
+            Entry<String, String> possibleValue = possibleValues.get(i);
+            printout.add(possibleValue.getKey() + "=" + possibleValue.getValue());
+         }
+
+         buildPrintout.add("Found ROS 2 property: " + printout);
+      }
+
+      return !possibleValues.isEmpty() ? possibleValues.peek().getValue() : null;
+   }
+
+   private int findDomainID()
+   {
+      int domainID = UNSET_DOMAIN_ID;
+
+      String valueForField = findValueForField("ROS_DOMAIN_ID", "ros.domain.id", "RTPSDomainID");
+
+      if (valueForField != null)
+      {
+         try
+         {
+            domainID = Integer.parseInt(valueForField.trim());
+         }
+         catch (NumberFormatException e)
+         {
+            LogTools.error("Unable to parse ROS Domain ID");
+         }
+      }
+
+      return domainID;
+   }
+
+   private InetAddress[] findAddressRestriction()
+   {
+      String valueForField = findValueForField("ROS_ADDRESS_RESTRICTION", "ros.address.restriction", "RTPSSubnet");
+
+      return convertToInetAddressArray(valueForField != null ? valueForField : "127.0.0.1/8");
+   }
+
+   protected static boolean domainIDValid(int domainID)
+   {
+      return domainID >= 0 && domainID <= 232;
+   }
+
+   /**
+    * Convert IP address CSV to InetAddress array
+    *
+    * @param restrictionHostString A list of IP addresses separated by comma in CIDR format. E.g. "127.0.0.1/8, 0.0.0.0/24"
+    * @return The array of InetAddresses representing the CSV list
+    */
+   protected static InetAddress[] convertToInetAddressArray(String restrictionHostString)
+   {
+      Set<InetAddress> foundAddressRestrictions = new HashSet<>();
+
+      String[] restrictionHostList = restrictionHostString.split("\\s*,\\s*");
+
+      List<InterfaceAddress> interfaceAddresses;
+      try
+      {
+         interfaceAddresses = Collections.list(NetworkInterface.getNetworkInterfaces())
+                                         .stream()
+                                         .flatMap(networkInterface -> networkInterface.getInterfaceAddresses().stream())
+                                         .collect(Collectors.toList());
```

Comment:
I think this is the same...
```suggestion
         interfaceAddresses = NetworkInterface.networkInterfaces().flatMap(networkInterface -> networkInterface.getInterfaceAddresses().stream()).toList();
```

Reply:
Yeah, that works